lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -54,7 +55,44 @@ def post(self):
 
             "start" and "end" are time strings representing a set of Elasticsearch
                 run document indices in which the dataset will be found.
+        """
+        user = json_data["user"]
+        name = json_data["name"]
+        start = json_data["start"]
+        end = json_data["end"]
+        self.logger.info(
+            "Return dataset {} for user {}, prefix {}: ({} - {})",
+            name,
+            user,
+            self.prefix,
+            start,
+            end,
+        )
 
+        # TODO: Need to refactor the template processing code from indexer.py
+        # to maintain the essential indexing information in a persistent DB
+        # (probably a Postgresql table) so that it can be shared here and by
+        # the indexer without re-loading on each access. For now, the index
+        # version is hardcoded.
+        uri_fragment = self._gen_month_range(".v6.run-data.", start, end)
```

Comment:
OK, `assemble()` is used slightly more broadly than I initially expected, so it might not be possible to make the use of `_gen_month_range()` common without inserting an intermediate superclass, which is probably more trouble than it is worth, especially if you can get rid of the first parameter, as I suggest elsewhere.  But, I still think it's a good idea to separate the path from the rest of the dictionary...how ever it makes the most sense to do so.

Reply:
Not all of the APIs use the `run-data` index. I could possibly pull that into the constructor, but I'm not sure I see the value right now. I'll see how things fall out when I implement the template DB table, which will change a lot.